src/java.base/share/classes/sun/util/resources/LocaleData.java

```diff
@@ -325,16 +324,16 @@ public String toBundleName(String baseName, Locale locale) {
         private static final SupplementaryStrategy INSTANCE
                 = new SupplementaryStrategy();
         // TODO: avoid hard-coded Locales
-        private static Set<Locale> JAVA_BASE_LOCALES
+        private final static Set<Locale> JAVA_BASE_LOCALES
             = Set.of(Locale.ROOT, Locale.ENGLISH, Locale.US);
 
         private SupplementaryStrategy() {
         }
 
         @Override
         public List<Locale> getCandidateLocales(String baseName, Locale locale) {
-            // Specifiy only the given locale
-            return Arrays.asList(locale);
+            // Specify only the given locale
+            return List.of(locale);
```

Comment:
Is it guaranteed that `locale` is not `null` here?

Reply:
This is a good question. As far I can see, the only call site checks for null explicitly: https://github.com/openjdk/jdk/blob/a4c46e1e4f4f2f05c8002b2af683a390fc46b424/src/java.base/share/classes/sun/util/resources/Bundles.java#L113. So the answer is yes.